### PR TITLE
Replace Clone() with View()

### PR DESCRIPTION
**Pull request by HieDean**
Replace `Clone()` with `View()`

### CODE REVIEW

**Comment by csukuangfj** [comment]
There are also other occurrences of `Clone()` that can be replaced with `View`. For instance,
https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-wenet-ctc-model.cc#L111-L112

https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-rnn-lm.cc#L74


The returned states are inputs to the neural network and won't be changed by the callers, so sharing the memory with `View()` should be safe.

**Comment by csukuangfj** [comment]
By the way,please fix the code stle issues
https://github.com/k2-fsa/sherpa-onnx/actions/runs/6900224733/job/18773046296?pr=432

You can check the code style locally by running
```bash
cd /path/to/sherpa-onnx
./scripts/check_style_cpplint.sh 2
```

**Comment by HieDean** [comment]
Follow your guide and make changes.

**Comment by HieDean** [comment]
For https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-wenet-ctc-model.cc#L111-L112 and
https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-rnn-lm.cc#L74
I leave them unchaged. I am not sure using `View()` in `GetInitStates()` is a good idea, because `GetInitStates()` is designed to have no right to modify member variables.

**Comment by csukuangfj** [comment]
> Follow your guide and make changes.

Thanks!


>  because GetInitStates() is designed to have no right to modify member variables.

Please change
https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-wenet-ctc-model.cc#L108
to
```
 std::vector<Ort::Value> GetInitStates() { 
```
by removing the trailing `const`.

If the CI passes, then it is safe to use `View` in `GetInitStates()`.

**Comment by HieDean** [comment]
For https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-wenet-ctc-model.cc#L111-L112, replace `Clone()` with `View()` is OK.
However, for https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-rnn-lm.cc#L74, each element of the second return of `GetInitStates()` will be moved at https://github.com/k2-fsa/sherpa-onnx/blob/1a6a41eb2c6570e67ca7f7f471683bc31698beab/sherpa-onnx/csrc/online-rnn-lm.cc#L34 in `Covert()`. It means next call of `GetInitStates()` will get nothing.

File: sherpa-onnx/csrc/online-lstm-transducer-model.cc [hunk 1]
```diff
@@ -250,7 +250,7 @@ Ort::Value OnlineLstmTransducerModel::RunDecoder(Ort::Value decoder_input) {
 Ort::Value OnlineLstmTransducerModel::RunJoiner(Ort::Value encoder_out,
                                                 Ort::Value decoder_out) {
   std::array<Ort::Value, 2> joiner_input = {std::move(encoder_out),
-                                            std::move(decoder_out)};
+                                            View(&decoder_out)};
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/online-lstm-transducer-model.cc]: By the way, we don't need to replace `move` with `View`.

File: sherpa-onnx/csrc/online-rnn-lm.cc [hunk 2]
```diff
@@ -67,13 +67,13 @@ class OnlineRnnLM::Impl {
...
+  std::pair<Ort::Value, std::vector<Ort::Value>> GetInitStates() {
     std::vector<Ort::Value> ans;
     ans.reserve(init_states_.size());
-    for (const auto &s : init_states_) {
-      ans.emplace_back(Clone(allocator_, &s));
+    for (auto &s : init_states_) {
+      ans.emplace_back(View(&s));
     }
-    return {std::move(Clone(allocator_, &init_scores_.value)), std::move(ans)};
+    return {std::move(View(&init_scores_.value)), std::move(ans)};
```
Comment (csukuangfj) [inline sherpa-onnx/csrc/online-rnn-lm.cc]:
```suggestion
    return {View(&init_scores_.value), std::move(ans)};
```

**Comment by csukuangfj** [comment]
Thank you for your contribution!

**Comment by HieDean** [comment]
Thank you for your guidance! I feel very happy that I can contribute to the community. And I have learned a lot during PR.

**Pull request merged**